atum/src/main/scala/za/co/absa/atum/utils/InfoFile.scala

```diff
@@ -22,30 +22,47 @@ import org.apache.hadoop.fs.{FileSystem, Path}
 import za.co.absa.atum.AtumImplicits.{DefaultControlInfoLoader, DefaultControlInfoStorer, StringPathExt}
 import za.co.absa.atum.location.S3Location.StringS3LocationExt
 
-private[atum] case class InfoFile(infoFile: String) {
-
-  private val validatedInfoFile: Option[String] = if (infoFile.isEmpty) None else Some(infoFile)
+object InfoFile {
+  /**
+   * Sanitizes (removes ?s and *s) and converts string full path to Hadoop FS and Path, e.g.
+   * `s3://mybucket1/path/to/file` -> S3 FS + `path/to/file`
+   * `/path/on/hdfs/to/file` -> local HDFS + `/path/on/hdfs/to/file`
+   *
+   * Note, that non-local HDFS paths are not supported in this method, e.g. hdfs://nameservice123:8020/path/on/hdfs/too.
+   *
+   * @param fullPath path to convert to FS and relative path
+   * @param hadoopConfiguration
+   * @return FS + relative path
+   */
+  def convertFullPathToFsAndRelativePath(fullPath: String)(implicit hadoopConfiguration: Configuration): (FileSystem, Path) = {
+    val sanitizedFullPath = fullPath.replaceAll("[\\*\\?]", "")
 
-  def toOptFsPath(implicit hadoopConfiguration: Configuration): Option[(FileSystem, Path)] = {
-    validatedInfoFile.map { definedInfoFile =>
-      definedInfoFile.toS3Location match {
+    sanitizedFullPath.toS3Location match {
 
-        case Some(s3Location) =>
-          // this is S3 over hadoop FS API, not SDK S3 approach
-          val s3Uri = new URI(s3Location.s3String) // s3://<bucket>
-          val s3Path = new Path(s"/${s3Location.path}") // /<text-file-object-path>
+      case Some(s3Location) =>
+        // this is S3 over hadoop FS API, not SDK S3 approach
+        val s3Uri = new URI(s3Location.s3String) // s3://<bucket>
+        val s3Path = new Path(s"/${s3Location.path}") // /<text-file-object-path>
 
-          val fs = FileSystem.get(s3Uri, hadoopConfiguration)
+        val fs = FileSystem.get(s3Uri, hadoopConfiguration)
 
-          (fs, s3Path)
+        (fs, s3Path)
 
-        case None => // hdfs location
-          val fs = FileSystem.get(hadoopConfiguration)
+      case None => // local hdfs location
+        val fs = FileSystem.get(hadoopConfiguration)
 
-          (fs, definedInfoFile.toPath)
-      }
+        (fs, sanitizedFullPath.toPath)
     }
   }
+}
+
+private[atum] case class InfoFile(infoFile: String) {
+
+  private val validatedInfoFile: Option[String] = if (infoFile.isEmpty) None else Some(infoFile)
```

Comment:
I know this is not @dk1844 code, but a remnant of the past. But why is it validated? Not sure I get it.

Reply:
validation here means that it is non-empty, because the original signature was written that empty string meant "no info file to be used". But your attention to this fact makes me question the need to stick with this flawed design -> I will probably deprecate it then and replace with Option[String] usage. Thanks for the hint.

Reply:
In view of this, as outlined, I have changed the `InfoFile` class (no more `validatedInfoFile`) and also `SparkSessionWrapper.enableControlMeasuresTracking` to work with `Option[String]`s now. The original version is kept and deprecated. 

Note, that it is the original, for deprecation-compatibility reason, that is tied to the default empty-param call `enableControlMeasuresTracking()`. When this gets removed, we can provide default params to the new version `(=None, = None)`.